chapters/chapter_5/section_5.md

```diff
@@ -236,118 +263,62 @@ dummy_dd_new(GlobalConfig *cfg)
 
   log_threaded_dest_driver_init_instance(&self->super, cfg);
   self->super.super.super.super.init = dummy_dd_init;
+  self->super.super.super.super.deinit = dummy_dd_deinit;
   self->super.super.super.super.free_fn = dummy_dd_free;
 
   self->super.worker.thread_init = dummy_worker_thread_init;
   self->super.worker.thread_deinit = dummy_worker_thread_deinit;
+  self->super.worker.connect = dummy_dd_connect;
   self->super.worker.disconnect = dummy_dd_disconnect;
   self->super.worker.insert = dummy_worker_insert;
 
   self->super.format_stats_instance = dummy_dd_format_stats_instance;
   self->super.super.super.super.generate_persist_name = dummy_dd_format_persist_name;
-  //self->super.stats_source = SCS_DUMMY;
+  self->super.stats_source = stats_register_type("dummy");
 
   return (LogDriver *)self;
 }
+```
 
-extern CfgParser dummy_parser;
-
-static Plugin dummy_plugin =
-{
-  .type = LL_CONTEXT_DESTINATION,
-  .name = "dummy",
-  .parser = &dummy_parser,
-};
-
-gboolean
-dummy_module_init(PluginContext *context, CfgArgs *args)
-{
-  plugin_register(context, &dummy_plugin, 1);
+`DummyDriver` is `LogThreadedDestDriver`. A `LogThreadedDestDriver` has at least one `LogThreadedDestWorker`. It is the worker's responsibility to send the messages.
 
-  return TRUE;
-}
-
-const ModuleInfo module_info =
-{
-  .canonical_name = "dummy",
-  .version = SYSLOG_NG_VERSION,
-  .description = "This is a dummy destination for syslog-ng.",
-  .core_revision = SYSLOG_NG_SOURCE_REVISION,
-  .plugins = &dummy_plugin,
-  .plugins_len = 1,
-};
-```
+You can have as many workers as you need, but for simplest cases one worker is enough. In that case, for the sake of simplicity, you can implement `LogThreadedDestDriver` to work in a so called `compatibility mode`. This is how `DummyDriver` is implemented in the example above. `LogThreadedDestDriver` embeds a `LogThreadedDestWorker` instance. In compatibility mode, `LogThreadedDestDriver` will use the embedded worker. If `compatibility mode` is not enough for you, then you need to override the `construct` method of `LogThreadedDestDriver`.
```

Comment:
I have to rethink this, so just I mention my initial opinion, I'll update it soon:
compat mode created for the "legacy" threaded-dest drivers, I don't know if we should encourage this for new drivers.

You could argue that compat mode is used and tested, and it would save the developer from some boiler-plate (maybe boiler-plate is too strong as an exprassion).